### PR TITLE
PP 64 student applications front

### PR DESCRIPTION
**Pull request by mariyador**
### Description

This update enhances the StudentNotifications component to display the student's application information and status. It allows students to decline their applications and provides styling for scenarios with no applications. Additionally, the component now navigates to the search classes page when no applications are present, and the route has been updated from notifications to applications. The label "Notifications" has been renamed to "Applications," and the related icon has been changed. 

### Changes
Functionality:
- Displayed student application information and status.
- Enabled students to decline their applications.
- Updated route from notifications to applications.

Styling:
- Added style for cases with no applications, including a button to navigate to the search classes page.
- Renamed "Notifications" to "Applications" and changed the icon.
- Adjusted profile picture sizes.

### Screenshots
Student applications:
Screenshot 2024-09-01 at 12 59 23 AM

Student decline applications:
Screenshot 2024-09-01 at 1 00 31 AM

Student with no applications:
Screenshot 2024-09-01 at 1 27 13 AM

### CODE REVIEW

**Review by vrudnitskaya — APPROVED** [review]
Awesome! I can book lesson and get the approve or decline back in real time!
decline
approve

File: src/components/pages/HomePage.jsx [hunk 1]
```diff
@@ -10,9 +10,14 @@ import { useNavigate } from 'react-router-dom';
 
 const HomePage = () => {
   const navigate = useNavigate();
-  const search = () => {
-    navigate('/search');
+  const search = (query) => {
+    if (query) {
+      navigate(`/search?query=${encodeURIComponent(query)}`);
```
Comment (vrudnitskaya) [inline src/components/pages/HomePage.jsx:15]: Thanks for adding this functionality.

**Review by AnnaPestova1 — APPROVED** [review]
Hi Mariya! 

I love this functionality! Sockets are awesome.

You did a great job here.

I have couple suggestions that you can add, decline or do in the separate ticket:
1. I really want to be redirected somewhere after as a student I applied to the lesson. Right now I am staying on the same page: 

https://github.com/user-attachments/assets/d29eda30-e205-4434-ba4b-1b1581b8af46

2. I Think we need confirmation window when we a approving or declining  the lesson (on both sides: as a teacher and as a student) 

3. can we add a number of messages in the application tab if we have something inside... or add any type of communication to user the tab has something inside 
image

**Comment by mariyador** [comment]
> Hi Mariya!
> 
> I love this functionality! Sockets are awesome.
> 
> You did a great job here.
> 
> I have couple suggestions that you can add, decline or do in the separate ticket:
> 
> 1. I really want to be redirected somewhere after as a student I applied to the lesson. Right now I am staying on the same page:
> 
>  C6EB3EC1-718C-495E-8848-8ABC5B683A2F.mov 
> 2. I Think we need confirmation window when we a approving or declining  the lesson (on both sides: as a teacher and as a student)
> 3. can we add a number of messages in the application tab if we have something inside... or add any type of communication to user the tab has something inside
> 
> image

Thank you, Anna!
Number of applications will be done in separate pr when i will be rewriting teachers' applications&
Student already receive a notification that he/she has been approved for class. May be i will add it for teacher to.
Redirecting... May be only if send the student to his applications page

**Pull request merged**